service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/AbstractEmailContentProvider.java

```diff
@@ -36,13 +39,11 @@ protected AbstractEmailContentProvider(String uiBaseUrl,
         this.hearingBookingService = hearingBookingService;
     }
 
-    @SuppressWarnings("unchecked")
-    ImmutableMap.Builder<String, Object> getCasePersonalisationBuilder(CaseDetails caseDetails, CaseData caseData) {
-        List<String> ordersAndDirections = buildOrdersAndDirections(
-            (Map<String, Object>) caseDetails.getData().get("orders"));
+    ImmutableMap.Builder<String, Object> getCasePersonalisationBuilder(Long caseId, CaseData caseData) {
+        List<String> ordersAndDirections = buildOrdersAndDirections(caseData.getOrders());
 
-        Optional<String> timeFrame = Optional.ofNullable((Map<String, Object>) caseDetails.getData().get("hearing"))
-            .map(hearing -> (String) hearing.get("timeFrame"));
+        Optional<String> timeFrame = Optional.ofNullable(caseData.getHearing()).map(
+            Hearing::getTimeFrame).filter(StringUtils::isNotBlank);
```

Comment:
minor, could be formatted as:
Optional.ofNullable(caseData.getHearing())
            .map(Hearing::getTimeFrame)
            .filter(StringUtils::isNotBlank)